### PR TITLE
integrate lightning into DM

### PR DESCRIPTION
**Pull request by lance6716**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):https://github.com/pingcap/docs/pull/12272

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Frank945946 — APPROVED** [review]

**Review by yudongusa — COMMENTED** [review]
We should make design doc in English. @niubell

File: dm/dm-precheck.md [hunk 1]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
 * 分表中自增主键检查
 
     - 分表存在自增主键时返回警告。如果存在自增主键冲突，请参照[自增主键冲突处理](/dm/shard-merge-best-practices.md#自增主键冲突处理)解决。
+    
+#### physical import 检查项
+
+在任务配置中使用 `import-mode: "logical"` 后，会增加如下的前置检查项以保证 physical import 的运行。如果这些前置检查难以解决，用户可以尝试使用 logical import 进行导入。
```
Comment (lance6716) [inline dm/dm-precheck.md:94]:
```suggestion
在任务配置中使用 `import-mode: "physical"` 后，会增加如下的前置检查项以保证 physical import 的运行。如果参照提示这些前置检查仍然难以解决，用户可以尝试使用 logical import 进行导入。
```

File: dm/dm-precheck.md [hunk 2]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+
+    - 统计不同的 TiKV 上的 Region 数目，如果最少的 Region 数目与最多的 Region 数目比例小于 0.75 会对用户返回警告。可以调整 PD 相关参数加快 Region 调度速度，并等待数目变化以解除警告
+    
+* 下游数据库 TiDB、PD、TiKV 组建版本检查
+
+    - physical import 需要调用 TiDB、PD、TiKV 接口，如果版本不符合要求会返回错误
+    
+* 下游数据库剩余空间检查
+
+    - 计算上游数据库所有白名单表的估算大小之和（`source_size`），如果下游数据库剩余空间小于 `source_size`，前置检查会返回错误；如果下游数据库剩余空间小于 `source_size` * TiKV 副本数 * 2，前置检查会返回警告
```
Comment (lance6716) [inline dm/dm-precheck.md:110]:
```suggestion
    - 计算上游数据库所有白名单表的估算大小之和（`source_size`），如果下游数据库剩余空间小于 `source_size`，前置检查会返回错误；如果下游数据库剩余空间小于 TiKV 副本数 *`source_size`*2，前置检查会返回警告
```

File: dm/task-configuration-file-full.md [hunk 3]
```diff
@@ -119,14 +119,28 @@ loaders:                             # load 处理单元的运行配置参数
...
+    # - "ignore"。保留已有数据，忽略新数据。
+    # - "error"。插入重复数据时报错并停止同步任务。
+    on-duplicate-logical: "replace"
+    # physical import 针对冲突数据的解决方式：
+    # - "none"。对应 TiDB Lightning physical import 冲突数据检测的 [none](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#冲突数据检测) 选项，表示遇到冲突数据时不进行处理。该模式性能最佳，但下游数据库会遇到数据索引不一致的问题。
+    # - "manual"。对应 TiDB Lightning physical import 冲突数据检测的 [remove](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#冲突数据检测) 选项。在遇到冲突数据时将所有相互冲突的数据删除，并记录在 ${meta-schema}_${name}.conflict_error_v1 表中。在本配置文件中，会记录在 dm_meta_test.conflict_error_v1 表中。全量导入阶段结束后，任务会暂停并提示用户查询这张表并按照[文档](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#冲突数据检测)进行手动处理。使用 resume-task 命令让任务恢复运行并进入到增量同步阶段。
+    on-duplicate-physical: "none"
+    # physical import 用作本地排序的目录位置，该选项的默认值与 dir 配置项一致。具体说明可以参见 [TiDB Lightning 必要条件及限制 - 存储空间](/tidb-lightning/tidb-lightning-physical-import-mode.md#必要条件及限制)的说明。
+    sorting-dir-physical: "./dumped_data"
+    # 磁盘空间限制，对应 [TiDB Lightning disk-quota 配置](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#磁盘资源配额-从-v620-版本开始引入)。具体说明可以参见上述连接。
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:136]:
```suggestion
    # 磁盘空间限制，对应 TiDB Lightning disk-quota 配置。具体说明参见文档：https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode-usage#磁盘资源配额-从-v620-版本开始引入

File: dm/task-configuration-file-full.md [hunk 4]
```diff
@@ -119,14 +119,28 @@ loaders:                             # load 处理单元的运行配置参数
...
+    # logical import 针对冲突数据的解决方式：
+    # - "replace"（默认值）。表示用最新数据替代已有数据。
+    # - "ignore"。保留已有数据，忽略新数据。
+    # - "error"。插入重复数据时报错并停止同步任务。
+    on-duplicate-logical: "replace"
+    # physical import 针对冲突数据的解决方式：
+    # - "none"。对应 TiDB Lightning physical import 冲突数据检测的 [none](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#冲突数据检测) 选项，表示遇到冲突数据时不进行处理。该模式性能最佳，但下游数据库会遇到数据索引不一致的问题。
+    # - "manual"。对应 TiDB Lightning physical import 冲突数据检测的 [remove](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#冲突数据检测) 选项。在遇到冲突数据时将所有相互冲突的数据删除，并记录在 ${meta-schema}_${name}.conflict_error_v1 表中。在本配置文件中，会记录在 dm_meta_test.conflict_error_v1 表中。全量导入阶段结束后，任务会暂停并提示用户查询这张表并按照[文档](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#冲突数据检测)进行手动处理。使用 resume-task 命令让任务恢复运行并进入到增量同步阶段。
+    on-duplicate-physical: "none"
+    # physical import 用作本地排序的目录位置，该选项的默认值与 dir 配置项一致。具体说明可以参见 [TiDB Lightning 必要条件及限制 - 存储空间](/tidb-lightning/tidb-lightning-physical-import-mode.md#必要条件及限制)的说明。
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:134]:
```suggestion
    # physical import 用作本地排序的目录位置，该选项的默认值与 dir 配置项一致。具体说明可以参见 TiDB Lightning 对存储空间的需求：https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode#运行环境需求
```

File: dm/dm-precheck.md [hunk 5]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
 
     - 分表存在自增主键时返回警告。如果存在自增主键冲突，请参照[自增主键冲突处理](/dm/shard-merge-best-practices.md#自增主键冲突处理)解决。
+    
+#### physical import 检查项
+
+在任务配置中使用 `import-mode: "physical"` 后，会增加如下的前置检查项以保证 physical import 的运行。如果参照提示这些前置检查仍然难以解决，用户可以尝试使用 logical import 进行导入。
+
+* 下游数据库空 Region 的检查
+
+    - 如果空 Region 的数量大于 `max(1000, 表的数量 * 3)`，即大于 "1000" 和 "3 倍表数量"二者中的较大者，会对用户返回警告。可以调整 PD 相关参数加快空 Region 的合并速度，并等待数目下降以解除警告
```
[the remark this answers is not in the document]
Comment (lance6716) [reply dm/dm-precheck.md:98]: https://docs.pingcap.com/zh/tidb/stable/pd-scheduling-best-practices#leaderregion-%E5%88%86%E5%B8%83%E4%B8%8D%E5%9D%87%E8%A1%A1

我先 commit 完已有的这些修改

File: dm/dm-precheck.md [hunk 6]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
 
     - 分表存在自增主键时返回警告。如果存在自增主键冲突，请参照[自增主键冲突处理](/dm/shard-merge-best-practices.md#自增主键冲突处理)解决。
+    
+#### Physical Import 检查项
+
+在任务配置中使用 `import-mode: "physical"` 后，会增加如下的前置检查项以保证 [Physical Import](/tidb-lightning/tidb-lightning-physical-import-mode.md) 正常运行。如果参照提示后仍然难以完成这些前置检查，你可以尝试使用 [Logical Import](/tidb-lightning/tidb-lightning-logical-import-mode.md) 进行导入。
+
+* 下游数据库空 Region 的检查
+
+    - 如果空 Region 的数量大于 `max(1000, 表的数量 * 3)`，即大于“1000”和“3 倍表数量”二者中的较大者，会对用户返回警告。可以调整 PD 相关参数加快空 Region 的合并速度，并等待数目下降以解除警告。
```
Comment (lance6716) [inline dm/dm-precheck.md:98]:
```suggestion
    - 如果空 Region 的数量大于 `max(1000, 表的数量 * 3)`，即大于“1000”和“3 倍表数量”二者中的较大者，会对用户返回警告。可以调整 PD 相关参数加快空 Region 的合并速度，并等待数目下降以解除警告。参见 [PD 调度策略最佳实践 - Region Merge 速度慢](best-practices/pd-scheduling-best-practices.md#region-merge-%E9%80%9F%E5%BA%A6%E6%85%A2)
```

File: dm/dm-precheck.md [hunk 7]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+    
+* 下游数据库 Region 分布的检查
+
+    - 统计不同的 TiKV 上的 Region 数目，如果最少的 Region 数目与最多的 Region 数目比例小于 0.75 会对用户返回警告。可以调整 PD 相关参数加快 Region 调度速度，并等待数目变化以解除警告
```
Comment (lance6716) [inline dm/dm-precheck.md:102]:
```suggestion
    - 统计不同的 TiKV 上的 Region 数目，如果最少的 Region 数目与最多的 Region 数目比例小于 0.75 会对用户返回警告。可以调整 PD 相关参数加快 Region 调度速度，并等待数目变化以解除警告。参见 [PD 调度策略最佳实践 - Leader/Region 分布不均衡](best-practices/pd-scheduling-best-practices.md#leaderregion-%E5%88%86%E5%B8%83%E4%B8%8D%E5%9D%87%E8%A1%A1)
```

**Review by ran-huang — APPROVED** [review]

File: dm/dm-precheck.md [hunk 8]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
 * 分表中自增主键检查
 
     - 分表存在自增主键时返回警告。如果存在自增主键冲突，请参照[自增主键冲突处理](/dm/shard-merge-best-practices.md#自增主键冲突处理)解决。
+    
+#### Physical Import 检查项
+
+在任务配置中使用 `import-mode: "physical"` 后，会增加如下的前置检查项以保证 [Physical Import](/tidb-lightning/tidb-lightning-physical-import-mode.md) 正常运行。如果参照提示后仍然难以完成这些前置检查，你可以尝试使用 [Logical Import](/tidb-lightning/tidb-lightning-logical-import-mode.md) 进行导入。
+
+* 下游数据库空 Region 的检查
```
Comment (TomShawn) [inline dm/dm-precheck.md:96]:
```suggestion
* 下游数据库中的空 Region
```

File: dm/dm-precheck.md [hunk 9]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+
+    - 如果空 Region 的数量大于 `max(1000, 表的数量 * 3)`，即大于“1000”和“3 倍表数量”二者中的较大者，会对用户返回警告。可以调整 PD 相关参数加快空 Region 的合并速度，并等待数目下降以解除警告。参见 [PD 调度策略最佳实践 - Region Merge 速度慢](/best-practices/pd-scheduling-best-practices.md#region-merge-速度慢)
```
Comment (TomShawn) [inline dm/dm-precheck.md:98]: not sure whether it is correct. pls take a look.
```suggestion
    - 如果空 Region 的数量大于 `max(1000, 表的数量 * 3)`，即大于“1000”和“3 倍表数量”二者中的较大者，前置检查会向用户返回警告。可以调整 PD 相关参数加快空 Region 的合并速度，并等待 Region 数量下降以解除警告。参见 [PD 调度策略最佳实践 - Region Merge 速度慢](/best-practices/pd-scheduling-best-practices.md#region-merge-速度慢)
```

File: dm/dm-precheck.md [hunk 10]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+    
+* 下游数据库 Region 分布的检查
```
Comment (TomShawn) [inline dm/dm-precheck.md:100]:
```suggestion
* 下游数据库中的 Region 分布
```

File: dm/dm-precheck.md [hunk 11]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+
+    - 统计不同的 TiKV 上的 Region 数目，如果最少的 Region 数目与最多的 Region 数目比例小于 0.75 会对用户返回警告。可以调整 PD 相关参数加快 Region 调度速度，并等待数目变化以解除警告。参见 [PD 调度策略最佳实践 - Leader/Region 分布不均衡](/best-practices/pd-scheduling-best-practices.md#leaderregion-分布不均衡)
```
Comment (TomShawn) [inline dm/dm-precheck.md:102]:
```suggestion
    - 统计不同的 TiKV 上的 Region 数目。如果最少的 Region 数目与最多的 Region 数目比例小于 0.75，前置检查会向用户返回警告。可以调整 PD 相关参数加快 Region 调度速度，并等待 Region 数目变化以解除警告。参见 [PD 调度策略最佳实践 - Leader/Region 分布不均衡](/best-practices/pd-scheduling-best-practices.md#leaderregion-分布不均衡)
```

File: dm/dm-precheck.md [hunk 12]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+    
+* 下游数据库 TiDB、PD、TiKV 组件版本检查
```
Comment (TomShawn) [inline dm/dm-precheck.md:104]:
```suggestion
* 下游数据库 TiDB、PD、TiKV 组件的版本
```

File: dm/dm-precheck.md [hunk 13]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+
+    - Physical Import 需要调用 TiDB、PD、TiKV 接口，如果版本不符合要求，会返回错误。
+    
+* 下游数据库剩余空间检查
```
Comment (TomShawn) [inline dm/dm-precheck.md:108]:
```suggestion
* 下游数据库的剩余空间
```

File: dm/dm-precheck.md [hunk 14]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+
+    - 计算上游数据库所有白名单表的估算大小之和 (`source_size`)，如果下游数据库剩余空间小于 `source_size`，前置检查会返回错误；如果下游数据库剩余空间小于 TiKV 副本数 \* `source_size` \* 2，前置检查会返回警告。
```
Comment (TomShawn) [inline dm/dm-precheck.md:110]:
```suggestion
    - 估算上游数据库所有白名单表的大小之和 (`source_size`)，如果下游数据库剩余空间小于 `source_size`，前置检查会返回错误；如果下游数据库剩余空间小于 TiKV 副本数 \* `source_size` \* 2，前置检查会返回警告。
```

File: dm/dm-precheck.md [hunk 15]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+    
+* 下游数据库是否在运行与 Physical Import 不兼容的任务
+
+    - 目前 Physical Import 不兼容 TiCDC、PITR 任务，如果发现下游数据库正在运行这些任务，前置检查会返回错误。
```
Comment (TomShawn) [inline dm/dm-precheck.md:114]: 是否需要给 TiCDC 和 PITR 加链接呢？

File: dm/dm-precheck.md [hunk 16]
```diff
@@ -130,6 +154,11 @@ tiup dmctl check-task ./task.yaml
 |schema_of_shard_tables|检查上游 MySQL 多实例分库分表的表结构一致性|
 |auto_increment_ID|检查上游 MySQL 多实例分库分表的自增主键冲突|
 |online_ddl|检查上游是否处于 [Online-DDL](/dm/feature-online-ddl.md) 过程中|
+|empty_region|Physical Import 空 Region 数目检查|
+|region_distribution|Physical Import Region 分布检查|
+|downstream_version|下游数据库 TiDB、PD、TiKV 版本检查|
+|free_space|下游数据库剩余空间检查|
+|downstream_mutex_features|下游数据库与 Physical Import 不兼容任务的检查|
```
Comment (TomShawn) [inline dm/dm-precheck.md:161]:
```suggestion
|empty_region|Physical Import 检查空 Region 的数目|
|region_distribution|Physical Import 检查 Region 的分布|
|downstream_version|检查下游数据库 TiDB、PD、TiKV 的版本|
|free_space|检查下游数据库的剩余空间|
|downstream_mutex_features|检查下游数据库是否存在与 Physical Import 不兼容的任务|
```

File: dm/monitor-a-dm-cluster.md [hunk 17]
```diff
@@ -62,15 +62,13 @@ overview 下包含运行当前选定 task 的所有 DM-worker/master instance/so
 
 | metric 名称 | 说明 | 告警说明 | 告警级别 |
 |:----|:------------|:----|:----|
+| dump progress | dump unit 导出过程的进度百分比，值变化范围为：0% - 100%  | N/A | N/A |
 | load progress | load unit 导入过程的进度百分比，值变化范围为：0% - 100%  | N/A | N/A |
-| data file size | load unit 导入的全量数据中数据文件（内含 `INSERT INTO` 语句）的总大小 | N/A | N/A |
+| checksum progress | load unit 导入完成后，数据校验过程的进度百分比，值变化范围为：0% - 100%  | N/A | N/A |
+| total bytes for load unit | load unit 导入过程中各组件处理的字节数，包括源数据解析、生成数据 KV、生成索引 KV 组件 | N/A | N/A |
```
Comment (TomShawn) [inline dm/monitor-a-dm-cluster.md:68]: 是导入过程包括源数据解析、生成数据 KV、生成索引 KV 组件吗？

**Review by TomShawn — COMMENTED** [review]
mostly minor comments

File: dm/dm-precheck.md (hunk 9 above)
Comment (ran-huang) [reply dm/dm-precheck.md:98]: "并等待 Region 数量下降以解除警告" -> “并等待**空** Region 数量下降以解除警告”？

File: dm/dm-precheck.md [hunk 18]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+
+    - Physical Import 需要调用 TiDB、PD、TiKV 接口，如果版本不符合要求，会返回错误。
+    
+* 下游数据库的剩余空间
+
+    - 估算上游数据库所有白名单表的大小之和 (`source_size`)，如果下游数据库剩余空间小于 `source_size`，前置检查会返回错误；如果下游数据库剩余空间小于 TiKV 副本数 \* `source_size` \* 2，前置检查会返回警告。
+    
+* 下游数据库是否在运行与 Physical Import 不兼容的任务
+
+    - 目前 Physical Import 不兼容 TiCDC、PITR 任务，如果发现下游数据库正在运行这些任务，前置检查会返回错误。
```
Comment (ran-huang) [inline dm/dm-precheck.md:114]:
```suggestion
    - 目前 Physical Import 不兼容 [TiCDC](/ticdc/ticdc-overview.md)、[PITR](/br/br-pitr-guide.md) 任务，如果发现下游数据库正在运行这些任务，前置检查会返回错误。
```

File: dm/dm-precheck.md (hunk 9 above)
Comment (TomShawn) [reply dm/dm-precheck.md:98]: OK

File: dm/monitor-a-dm-cluster.md (hunk 17 above)
Comment (lance6716) [inline dm/monitor-a-dm-cluster.md:68]:
```suggestion
| total bytes for load unit | load unit 导入过程中源数据解析、生成数据 KV、生成索引 KV 组件处理的字节数 | N/A | N/A |
```

Comment (lance6716) [reply dm/monitor-a-dm-cluster.md:68]: @TomShawn PTAL

Comment (TomShawn) [reply dm/monitor-a-dm-cluster.md:68]: LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

File: dm/dm-precheck.md [hunk 19]
```diff
@@ -88,6 +88,30 @@ tiup dmctl check-task ./task.yaml
...
+
+在任务配置中使用 `import-mode: "physical"` 后，会增加如下的前置检查项以保证 [Physical Import](/tidb-lightning/tidb-lightning-physical-import-mode.md) 正常运行。如果参照提示后仍然难以完成这些前置检查，你可以尝试使用 [Logical Import](/tidb-lightning/tidb-lightning-logical-import-mode.md) 进行导入。
+
+* 下游数据库中的空 Region
+
+    - 如果空 Region 的数量大于 `max(1000, 表的数量 * 3)`，即大于“1000”和“3 倍表数量”二者中的较大者，前置检查会向用户返回警告。可以调整 PD 相关参数加快空 Region 的合并速度，并等待空 Region 数量下降以解除警告。参见 [PD 调度策略最佳实践 - Region Merge 速度慢](/best-practices/pd-scheduling-best-practices.md#region-merge-速度慢)
+    
+* 下游数据库中的 Region 分布
+
+    - 统计不同的 TiKV 上的 Region 数目。如果最少的 Region 数目与最多的 Region 数目比例小于 0.75，前置检查会向用户返回警告。可以调整 PD 相关参数加快 Region 调度速度，并等待 Region 数目变化以解除警告。参见 [PD 调度策略最佳实践 - Leader/Region 分布不均衡](/best-practices/pd-scheduling-best-practices.md#leaderregion-分布不均衡)
```
Comment (TomShawn) [inline dm/dm-precheck.md:102]: I am not sure whether my comment is correct. PTAL
```suggestion
    - 统计不同的 TiKV 上的 Region 数目。假设 Region 数最少的 TiKV 节点上拥有 `a` 个 Region，Region 数最多的 TiKV 节点上拥有 `b` 个 Region，如果 a/b 小于 0.75，则前置检查会向用户返回警告。可以调整 PD 相关参数加快 Region 调度速度，并等待 Region 数目变化以解除警告。参见 [PD 调度策略最佳实践 - Leader/Region 分布不均衡](/best-practices/pd-scheduling-best-practices.md#leaderregion-分布不均衡)
```

File: dm/task-configuration-file-full.md [hunk 20]
```diff
@@ -119,14 +119,35 @@ loaders:                             # load 处理单元的运行配置参数
...
+    # - "logical"(默认)。使用 TiDB Lightning logical import 进行导入。文档：https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-logical-import-mode
+    # - "physical"。使用 TiDB Lightning physical import 进行导入。文档：https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode
+    import-mode: "logical"
+    # logical import 针对冲突数据的解决方式：
+    # - "replace"（默认值）。表示用最新数据替代已有数据。
+    # - "ignore"。保留已有数据，忽略新数据。
+    # - "error"。插入重复数据时报错并停止同步任务。
+    on-duplicate-logical: "replace"
+    # physical import 针对冲突数据的解决方式：
+    # - "none"。对应 TiDB Lightning physical import 冲突数据检测的 "none" 选项
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:131]:
```suggestion
    # - "none"（默认）。对应 TiDB Lightning physical import 冲突数据检测的 "none" 选项 
```

File: dm/dm-precheck.md (hunk 19 above)
Comment (lance6716) [reply dm/dm-precheck.md:102]: lgtm

File: dm/task-configuration-file-full.md [hunk 21]
```diff
@@ -119,14 +119,35 @@ loaders:                             # load 处理单元的运行配置参数
...
-    # - "sql"(默认)。使用 [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md) TiDB-backend 进行导入。
-    # - "loader"。使用 Loader 导入。此模式仅作为兼容模式保留，目前用于支持 TiDB Lightning 尚未包含的功能，预计会在后续的版本废弃。
-    import-mode: "sql"
-    # 全量导入阶段针对冲突数据的解决方式：
-    # - "replace"（默认值）。仅支持 import-mode 为 "sql"，表示用最新数据替代已有数据。
-    # - "ignore"。仅支持 import-mode 为 "sql"，保留已有数据，忽略新数据。
-    # - "error"。仅支持 import-mode 为 "loader"。插入重复数据时报错并停止同步任务。
-    on-duplicate: "replace"
+    # - "logical"(默认)。使用 TiDB Lightning logical import 进行导入。文档：https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-logical-import-mode
+    # - "physical"。使用 TiDB Lightning physical import 进行导入。文档：https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:123]:
```suggestion
    # - "physical"。使用 TiDB Lightning physical import 进行导入。文档：https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode
    #   当前 "physical" 为实验特性，不建议在生产环境中使用。
```

**Pull request merged**